src/nwb2bids/bids_models/_participant.py

```diff
@@ -52,12 +54,31 @@ def from_nwbfiles(cls, nwbfiles: list[pydantic.InstanceOf[pynwb.NWBFile]]) -> ty
             raise NotImplementedError(message)
 
         nwbfile = nwbfiles[0]
+        participant_id = nwbfile.subject.subject_id.replace("_", "-")
+
+        species = nwbfile.subject.species
+        if species is not None and not re.match(pattern=_SPECIES_REGEX, string=species):
+            message = (
+                f"Subject species '{species}' within NWB file is not a proper Latin binomial or NCBI Taxonomy link. "
+                "Skipping automated extraction."
+            )
+            warnings.warn(message=message, stacklevel=2)
+            species = None
+
+        sex = nwbfile.subject.sex
+        if nwbfile.subject.sex is not None and nwbfile.subject.sex not in _ALLOWED_SEXES:
+            message = (
+                f"Subject sex '{sex}' within NWB file is not one of the allowed patterns. "
+                "Skipping automated extraction."
+            )
+            warnings.warn(message=message, stacklevel=2)
+            sex = None
```

Comment:
These work using Pydantic field validators or annotating with type metadata may be cleaner. That approach can potentially eliminate some of the duplications in `_ALLOWED_SEXES`.

I wouldn't try to improve these in this PR though. If you file another issue, I may give it a shot.

Reply:
Yeah, the error collection improvement will be a followup PR(s) and will impact all places where a potential 'improvement' (describing it gently) could happen